lucet-runtime/lucet-runtime-internals/src/vmctx.rs

```diff
@@ -363,6 +364,16 @@ impl Vmctx {
             val: YieldedVal::new(val),
             expecting: expecting as Box<dyn Any>,
         };
+
+        let terminable = inst.kill_state.terminable.swap(false, Ordering::SeqCst);
+
+        if !terminable {
```

Comment:
This and the small `jz _hang` loop in `context_asm.S` are two places I think it's best to just spin for a moment - if a processor stays in them for even microseconds I'd be surprised and concerned, but it's entirely dependent on how long signal delivery takes.

Reply:
Will our demise ever arrive? This is only called from a hostcall, so the execution domain shouldn't allow a signal to be sent. I think we instead need to continue on to swap back to the host, but then check for `Domain::Terminated` on the host side.